kratos/solving_strategies/builder_and_solvers/residualbased_block_builder_and_solver.h

```diff
@@ -1327,6 +1351,86 @@ class ResidualBasedBlockBuilderAndSolver
         }
     }
 
+    /**
+     * @brief This method returns the scale norm considering for scaling the diagonal
+     * @param rA The LHS matrix
+     * @param rCurrentProcessInfo The current process info instance
+     * @param rScaleVariable The scale variable considered
+     * @return The scale norm
+     */
+    double GetScaleNorm(
+        TSystemMatrixType& rA,
+        const ProcessInfo& rCurrentProcessInfo,
+        const Variable<double>& rScaleVariable
+        )
+    {
+        if (mOptions.Is(SCALE_DIAGONAL) ) {
+            if (rCurrentProcessInfo.Has(rScaleVariable)) {
+                return rCurrentProcessInfo[rScaleVariable];
+            } else {
+                if (mOptions.Is(CONSIDER_NORM_DIAGONAL) ) {
+                    return GetDiagonalNorm(rA);
+                } else {
+                    return TSparseSpace::TwoNorm(rA);
+                }
+            }
+        } else {
+            return 1.0;
+        }
+    }
+
+    /**
+     * @brief This method returns the diagonal norm considering for scaling the diagonal
+     * @param rA The LHS matrix
+     * @return The diagonal norm
+     */
+    double GetDiagonalNorm(TSystemMatrixType& rA)
+    {
+        double diagonal_norm = 0.0;
+        for(IndexType i = 0; i < TSparseSpace::Size1(rA); ++i) {

```

Comment:
should be done in parallel with a reduction

Reply:
I am scared of the reduction and MSVC

Reply:
+ reduction works in msvc

Reply:
last post should be "+ reduction" works in MSVC

Reply:
Yeh, but not for Max